### PR TITLE
Get all 2.0 templates to basically work with the new Program.cs and Startup.cs formats and 2.0.0-* packages

### PR DESCRIPTION
**Pull request by mlorbetske**
cc @glennc

### CODE REVIEW

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-CSharp/Program.cs [hunk 1]
```diff
@@ -14,11 +15,8 @@ public static void Main(string[] args)
             var host = new WebHostBuilder()
                 .UseKestrel()
                 .UseContentRoot(Directory.GetCurrentDirectory())
-                .UseIISIntegration()
```
Comment (glennc) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-CSharp/Program.cs]: I don't think we can remove this yet, not until we are confident that the new ANCM has been shipped everywhere. @shirhatti

**Comment by glennc** [comment]
We need to update the Auth parts of this as well right, should that be in this PR or a different one?

**Comment by mlorbetske** [comment]
@glennc we'd want to add the auth stuff to this one I think

**Comment by glennc** [comment]
@HaoK isn't back until Wed, @danroth27 can you update these to the new system? or perhaps @blowdart

**Comment by blowdart** [comment]
I'm not sure what Hao changed,

**Comment by glennc** [comment]
I can go and look at all the samples and changes, get something happening that Hao can review, if nobody has been following along with the changes.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-CSharp/Program.cs (hunk 1 above)
Comment (shirhatti) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/EmptyWeb-CSharp/Program.cs]: Yeah. We can't remove this just yet.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs [hunk 2]
```diff
@@ -14,14 +16,31 @@ public static void Main(string[] args)
             var host = new WebHostBuilder()
                 .UseKestrel()
                 .UseContentRoot(Directory.GetCurrentDirectory())
-                .UseIISIntegration()
-                .UseStartup<Startup>()
-#if (IncludeApplicationInsights)
-                .UseApplicationInsights()
+                .ConfigureConfiguration((context, configBuilder) => {
+#if (IndividualAuth || OrganizationalAuth)
+                    configBuilder.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
```
Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs]: Should the `AddJsonFile` on this line move down to the next line, and be indented? Then the next line's `AddJsonFile` should also be indented.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs [hunk 3]
```diff
@@ -14,14 +16,31 @@ public static void Main(string[] args)
...
+
+                    if (context.HostingEnvironment.IsDevelopment())
+                    {
+                        // For more details on using the user secret store see https://go.microsoft.com/fwlink/?LinkID=532709
+                        builder.AddUserSecrets<Startup>();
+                    }
+
+                    configBuilder.AddEnvironmentVariables();
+#else
+                    configBuilder.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
```
Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Program.cs]: Move AddJsonFile to next line, and indent all the `.AddXyz`?

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Program.cs [hunk 4]
```diff
@@ -15,11 +16,28 @@ public static void Main(string[] args)
             var host = new WebHostBuilder()
                 .UseKestrel()
                 .UseContentRoot(Directory.GetCurrentDirectory())
-                .UseIISIntegration()
-                .UseStartup<Startup>()
-#if (IncludeApplicationInsights)
-                .UseApplicationInsights()
+                .ConfigureConfiguration((context, configBuilder) => {
+#if (OrganizationalAuth)
+                    configBuilder.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
```
Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Program.cs]: See earlier comments.

File: template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Program.cs [hunk 5]
```diff
@@ -15,11 +16,28 @@ public static void Main(string[] args)
...
+
+                    if (context.HostingEnvironment.IsDevelopment())
+                    {
+                        // For more details on using the user secret store see https://go.microsoft.com/fwlink/?LinkID=532709
+                        builder.AddUserSecrets<Startup>();
+                    }
+
+                    configBuilder.AddEnvironmentVariables();
+#else
+                    configBuilder.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
```
Comment (Eilon) [inline template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/WebApi-CSharp/Program.cs]: See earlier comments.

**Comment by Eilon** [comment]
@haok's changes aren't in yet and won't be until sometime later this week at best, so it might be better to just get these changes in, and then do the auth changes later.

**Pull request merged**